src/syrupy/session.py

```diff
@@ -28,25 +29,42 @@ class SnapshotSession:
     _invocation_args: Tuple[str, ...] = attr.ib(factory=tuple)
     report: Optional["SnapshotReport"] = attr.ib(default=None)
     # All the collected test items
-    _all_items: Dict["pytest.Item", bool] = attr.ib(factory=dict)
+    _collected_items: Set["pytest.Item"] = attr.ib(factory=set)
     # All the selected test items. Will be set to False until the test item is run.
-    _ran_items: Dict["pytest.Item", bool] = attr.ib(factory=dict)
+    _selected_items: Dict["pytest.Item", bool] = attr.ib(factory=dict)
+    # Used to optimise the look up of selected test items when marking them as ran.
+    _selected_items_by_nodeid: Dict[str, "pytest.Item"] = attr.ib(
+        factory=dict, init=False
+    )
```

Comment:
Technically all selected items should already be in collected, could probably do away with the duplication?